haigha/transports/socket_transport.py

```diff
@@ -29,26 +34,33 @@ def connect(self, (host, port), klass=socket.socket):
         Connect assuming a host and port tuple.
         '''
         self._host = "%s:%s" % (host, port)
-        self._sock = klass()
-        self._sock.setblocking(True)
-        self._sock.settimeout(self.connection._connect_timeout)
-        if self.connection._sock_opts:
-            for k, v in self.connection._sock_opts.iteritems():
-                family, type = k
-                self._sock.setsockopt(family, type, v)
-        infos = socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP)
-        exc = socket.error("getaddrinfo returns an empty list")
-        for _family, _socktype, _proto, _canonname, sockaddr in infos:
+
+        for info in socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP):
+
+            family, socktype, proto, _, sockaddr = info
+            self._sock = klass(family, socktype, proto)
+            self._sock.setblocking(True)
```

Comment:
Let's remove the `setblocking` call. I don't know why the legacy code called `setblocking`, but it's unnecessary when followed up by `settimeout`,  because `settimeout` is a superset of `setblocking`: depending on the value of its parameter, `settimeout` will either make the socket blocking or non-blocking, in additional to recording the timeout value.